default.go

```diff
@@ -84,7 +87,7 @@ func (d *DefaultNode) setDefaults(fields models.Fields, tags models.Tags) (model
 	newTags := tags
 	tagsCopied := false
 	for tag, value := range d.d.Tags {
-		if _, ok := tags[tag]; !ok {
+		if v := tags[tag]; v == "" {
```

Comment:
While it is breaking, I think that this is the correct functionality. The reason being that it's not possible to write a point in line protocol with an empty tag value and therefore and points that would have meet this condition would be points where someone grouped by a tag that did not exist. If I'm a user, I would have to know when I group by something that doesn't exist, that the resulting data will be tagged with the non existent tag, but with an empty string and so I much use the `defaultEmptyTags()`. 

The counter argument to this is that it is a breaking change, and we shouldn't break things that people may be relying on.

I'm happy to go whichever route you think is best.